### DISCUSSION TITLE
How to combine serverless functions

### DISCUSSION THREAD

**Discussion by fromcafe** [vercel/community#3495] [Help] [answered]
### Problem Description

Is this feasible to use multiple functions with different advantages in a project? 
For example, using them,nodejs io, rust intensive computing, using py's gis toolkit, and configuring three types of runtimes in a vercel project.

Another question, does the serverless function have to be placed in the API/ directory? I tried placing it in a custom directory, such as src/fn/some_ functions.rs, but I failed.

```
# src app directory
# vercel config
{
  "functions": {
    "src/fn/**/*.rs": {
      "runtime": "vercel-rust@4.0.2"
    }
  }
}

# vercel error log
Error: The pattern "src/fn/**/*.rs" defined in `functions` doesn't match any Serverless Functions.
```

**Comment by amyegan** [comment, marked as answer]
Hey @fromwhite. You can combine different [Serverless Function Runtimes](https://vercel.com/docs/concepts/functions/serverless-functions/runtimes#) in your project. We support [Node.js](https://vercel.com/docs/concepts/functions/serverless-functions/runtimes/node-js), [Go](https://vercel.com/docs/concepts/functions/serverless-functions/runtimes/go), [Python](https://vercel.com/docs/concepts/functions/serverless-functions/runtimes/python), and [Ruby](https://vercel.com/docs/concepts/functions/serverless-functions/runtimes/ruby) in the `/api` directory with no additional config required in most cases. If additional configuration is needed for a particular situation, the function can be configured using the [`functions` property](https://vercel.com/docs/concepts/projects/project-configuration#functions) in `vercel.json`.

We also have community-built [Bash](https://github.com/importpw/vercel-bash), [Deno](https://github.com/vercel-community/deno), [PHP](https://github.com/vercel-community/php), and [Rust](https://github.com/vercel-community/rust) runtimes. These are not supported by default, but can be used by setting the [`functions` property](https://vercel.com/docs/concepts/projects/project-configuration#functions) in `vercel.json`.
